census/estimator/trainer/task.py

```diff
@@ -0,0 +1,162 @@
+    eval_input = model.generate_input_fn(
+        eval_file,
+        batch_size=eval_batch_size,
+        skip_header_lines=1,
+        shuffle=False
+    )
+    if args.serving_input_fn_type == 'JSON':
+      serving_input_fn = model.json_serving_input_fn
+    else:
+      serving_input_fn = model.tfrecord_serving_input_fn
+    return learn.Experiment(
+        model.build_estimator(
+            job_dir,
+            embedding_size=embedding_size,
+            hidden_units=hidden_units
+        ),
+        train_input_fn=train_input,
+        eval_input_fn=eval_input,
+        eval_metrics={
+            'training/hptuning/metric': learn.MetricSpec(
+                metric_fn=metrics.streaming_accuracy,
+                prediction_key='logits'
+            )
+        },
+        export_strategies=[saved_model_export_utils.make_export_strategy(
+
+            default_output_alternative_key=None,
+            exports_to_keep=1
+        )],
+        **experiment_args
+    )
+  return _experiment_fn
+
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser()
+  # Prediction Arguments
+  parser.add_argument(
+      '--serving-input-fn-type',
+      help='What signature to support in the exported model.',
+      choices=['JSON', 'TFRECORDS'],
+      default='JSON'
+  )
+  # Input Arguments
+  parser.add_argument(
+      '--train-file',
+      help='GCS or local path to training data',
+      required=True
+  )
+  parser.add_argument(
+      '--num-epochs',
+      help="""\
+      Maximum number of training data epochs on which to train.
+      If both --max-steps and --num-epochs are specified,
+      the training job will run for --max-steps or --num-epochs,
+      whichever occurs first. If unspecified will run for --max-steps.\
+      """,
+      type=int,
+  )
+  parser.add_argument(
+      '--train-batch-size',
+      help='Batch size for training steps',
+      type=int,
+      default=40
+  )
+  parser.add_argument(
+      '--eval-batch-size',
+      help='Batch size for evaluation steps',
+      type=int,
+      default=40
+  )
+  parser.add_argument(
+      '--train-steps',
+      help="""\
+      Steps to run the training job for. If --num-epochs is not specified,
+      this must be. Otherwise the training job will run indefinitely.\
+      """,
+      type=int
+  )
+  parser.add_argument(
+      '--eval-steps',
+      help='Number of steps to run evalution for at each checkpoint',
+      default=100,
+      type=int
+  )
+  parser.add_argument(
+      '--eval-file',
+      help='GCS or local path to evaluation data',
+      required=True
+  )
+  # Training arguments
+  parser.add_argument(
+      '--embedding-size',
+      help='Number of embedding dimensions for categorical columns',
+      default=8,
+      type=int
+  )
+  parser.add_argument(
+      '--hidden-units',
+      help='List of hidden layer sizes to use for DNN feature columns',
+      nargs='+',
+      type=int,
+      default=[100, 70, 50, 25]
```

Comment:
HP parameters must be unique, so --layer_size 100 --layer_size 70 does not work.

For allowing (--layer_size1 100  --layer_size2 70 --layer_size3 50 ....), I also used argparse. See line 280 in https://github.com/googledatalab/pydatalab/blob/cloudml/solutionbox/structured_data/datalab_solutions/structured_data/trainer/task.py